tools/apiview/emitters/cadl-apiview/src/apiview.ts

```diff
@@ -0,0 +1,739 @@
+import {
+  ArrayExpressionNode,
+  BaseNode,
+  BooleanLiteralNode,
+  DecoratorExpressionNode,
+  EnumMemberNode,
+  EnumSpreadMemberNode,
+  EnumStatementNode,
+  IdentifierNode,
+  InterfaceStatementNode,
+  IntersectionExpressionNode,
+  MemberExpressionNode,
+  ModelExpressionNode,
+  ModelPropertyNode,
+  ModelSpreadPropertyNode,
+  ModelStatementNode,
+  Namespace,
+  navigateProgram,
+  NumericLiteralNode,
+  OperationSignatureDeclarationNode,
+  OperationSignatureReferenceNode,
+  OperationStatementNode,
+  Program,
+  StringLiteralNode,
+  SyntaxKind,
+  TupleExpressionNode,
+  TypeReferenceNode,
+  UnionExpressionNode,
+} from "@cadl-lang/compiler";
+import { ApiViewDiagnostic, ApiViewDiagnosticLevel } from "./diagnostic.js";
+import { getFullyQualifiedNamespace } from "./emitter.js";
+import { ApiViewNavigation } from "./navigation.js";
+import { generateId, NamespaceModel } from "./namespace-model.js";
+import { LIB_VERSION } from "./version.js";
+
+export const enum ApiViewTokenKind {
+  Text = 0,
+  Newline = 1,
+  Whitespace = 2,
+  Punctuation = 3,
+  Keyword = 4,
+  LineIdMarker = 5, // use this if there are no visible tokens with ID on the line but you still want to be able to leave a comment for it
+  TypeName = 6,
+  MemberName = 7,
+  StringLiteral = 8,
+  Literal = 9,
+  Comment = 10,
+  DocumentRangeStart = 11,
+  DocumentRangeEnd = 12,
+  DeprecatedRangeStart = 13,
+  DeprecatedRangeEnd = 14,
+  SkipDiffRangeStart = 15,
+  SkipDiffRangeEnd = 16,
+}
+
+export interface ApiViewToken {
+  Kind: ApiViewTokenKind;
+  Value?: string;
+  DefinitionId?: string;
+  NavigateToId?: string;
+  CrossLanguageDefinitionId?: string;
+}
+
+let indentString: string = "";
+const indentSize: number = 2;
+
+export class ApiViewDocument {
+  Name: string;
+  PackageName: string;
+  Tokens: ApiViewToken[] = [];
+
+  Navigation: ApiViewNavigation[] = [];
+  Diagnostics: ApiViewDiagnostic[] = [];
```

Comment:
Is this needed, can't use the cadl diagnostic system?

Reply:
diagnostics in this case are an APIView feature and need to be part of the token file. The implementation is incomplete though since I don't foresee the need for Cadl to display any.

Reply:
Actually, speaking with Johan, there may be a use case for APIView's diagnostic to highlight which linter rules are suppressed. But for now, I'll take it as a follow-up item.